fedimint-server/src/consensus/transaction.rs

```diff
@@ -19,6 +20,21 @@ pub async fn process_transaction_with_dbtx(
         CONSENSUS_TX_PROCESSED_OUTPUTS.observe(out_count as f64);
     });
 
+    // We can not return the error here as errors are not returned in a specified
+    // order and the client still expects consensus on the error. Since the
+    // error is not extensible at the moment we need to incorrectly return the
+    // InvalidWitnessLength variant.
+    transaction
+        .inputs
+        .clone()
+        .into_par_iter()
+        .try_for_each(|input| {
+            modules
+                .get_expect(input.module_instance_id())
+                .verify_input(&input, input.module_instance_id())
```

Comment:
The way we use the `module_instance_id` argument here is absolutely useless (could just call that fn inside `verify_input`).